operator/webhook/addons/certmanager/main.py

```diff
@@ -0,0 +1,156 @@
+import logging
+from typing import Mapping, List
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _new_certificate(obj) -> Mapping:
+    metadata = obj["metadata"]
+
+    name = metadata["name"]
+
+    namespace = metadata["namespace"]
+
+    annotations = metadata.get("annotations", None)
+    if annotations is None:
+        _LOGGER.debug(
+            "IntegrationRoute does not contain metadata.annotations. No certificate will be generated."
+        )
+        return {}
+
+    cert_manager_io_annotations = [
+        annotation for annotation in annotations if "cert-manager.io" in annotation
+    ]
+
+    if cert_manager_io_annotations:
+        cluster_issuer = annotations.get("cert-manager.io/cluster-issuer", None)
+        if cluster_issuer is None:
+            _LOGGER.error(
+                "IntegrationRoute does not contain metadata.annotations.cert-manager.io/cluster-issuer"
+            )
+            return {}
+
+        common_name = annotations.get("cert-manager.io/common-name", None)
+        if common_name is None:
+            _LOGGER.error(
+                "IntegrationRoute does not contain metadata.annotations.cert-manager.io/common-name"
+            )
+            return {}

```

Comment:
:question: Instead of treating this as an error, does it make sense to use the integration-route's name as a default for `common_name`?

Reply:
Yes, Changed